### PR TITLE
NEW: leave requests: add field into type dictionary to block request if balance is negative

### PR DESCRIPTION
**Pull request by marc-dll**

### CODE REVIEW

File: htdocs/holiday/class/holiday.class.php [hunk 1]
```diff
@@ -259,6 +259,17 @@ public function create($user, $notrigger = 0)
 			$this->error = "ErrorBadParameterFkType"; return -1;
 		}
 
+		$checkBalance = getDictvalue(MAIN_DB_PREFIX.'c_holiday_types', 'block_if_negative', $this->fk_type);
+
+		if ($checkBalance > 0) {
```
Comment (eldy) [inline htdocs/holiday/class/holiday.class.php:264]: We must check when we validate only. 
But when we create, we must check only if status is not draft.

**Comment by marc-dll** [comment]
@eldy PR updated (finally!) following your remarks, and with conflicts solved

**Comment by marc-dll** [comment]
@eldy Conflicts solved again ;)

**Comment by marc-dll** [comment]
@eldy Conflicts solved another time. Can you please make a feedback if you consider it should not be merged, please ?

**Pull request merged**